### PR TITLE
[Merged by Bors] - feat(RingTheory/Powerseries): generalize `coeff_one_pow`

### PR DESCRIPTION
**Pull request by rwst**
Removes the condition of `coeff_one_pow` that the constant coefficient must be 1.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/noshake.json [hunk 1]
```diff
@@ -302,12 +305,11 @@
   "Mathlib.Lean.Expr.ExtraRecognizers": ["Mathlib.Data.Set.Defs"],
   "Mathlib.Lean.Expr.Basic": ["Batteries.Logic"],
   "Mathlib.Init.Logic": ["Batteries.Logic"],
-  "Mathlib.Logic.Function.Defs": ["Mathlib.Tactic.AdaptationNote"],
```
Comment (riccardobrasca) [inline scripts/noshake.json:305]: Are these changes wanted?

Comment (rwst) [reply scripts/noshake.json:305]: These were made by `lake exe shake --fix`.

Comment (riccardobrasca) [reply scripts/noshake.json:305]: Meaning that `shake` complains without them?

**Comment by riccardobrasca** [comment]
Can you add back the previous theorem, as a special case of the new result?

File: Mathlib/RingTheory/PowerSeries/Basic.lean [hunk 2]
```diff
@@ -690,18 +692,35 @@ lemma coeff_one_mul (φ ψ : R⟦X⟧) : coeff R 1 (φ * ψ) =
     mul_comm, add_comm]
   norm_num
 
-/-- First coefficient of the `n`-th power of a power series with constant coefficient 1. -/
-lemma coeff_one_pow (n : ℕ) (φ : R⟦X⟧) (hC : constantCoeff R φ = 1) :
-    coeff R 1 (φ ^ n) = n * coeff R 1 φ := by
+/-- First coefficient of the `n`-th power of a power series. -/
+lemma coeff_one_pow (n : ℕ) (hn : 1 ≤ n) (φ : R⟦X⟧) :
```
Comment (AntoineChambert-Loir) [inline Mathlib/RingTheory/PowerSeries/Basic.lean:696]: It seems that theorem holds if n = 0 (for ridiculous reasons) : both sides are zero.

Comment (riccardobrasca) [reply Mathlib/RingTheory/PowerSeries/Basic.lean:696]: Even better then!

**Comment by AntoineChambert-Loir** [comment]
Is it really needed, or cannot one just apply this result and go one with rewriting the constantCoeff and using `one_pow`?

**Comment by riccardobrasca** [comment]
> Is it really needed, or cannot one just apply this result and go one with rewriting the constantCoeff and using `one_pow`?

Hmm, you're right, it's not that natural and we can live without it.

**Comment by rwst** [comment]
Noted and closed, thanks.

**Comment by riccardobrasca** [comment]
I think Antoine was answering my comment, saying there is no need to add the current result back. Your generalization looks totally fine.

File: scripts/noshake.json (hunk 1 above)
Comment (rwst) [reply scripts/noshake.json:305]: OK, it builds without that line.

File: Mathlib/RingTheory/PowerSeries/Basic.lean [hunk 3]
```diff
@@ -581,18 +583,35 @@ lemma coeff_one_mul (φ ψ : R⟦X⟧) : coeff R 1 (φ * ψ) =
...
+    coeff R 1 (φ ^ n) = n * coeff R 1 φ * (constantCoeff R φ) ^ (n - 1) := by
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/PowerSeries/Basic.lean:588]:
```suggestion
/-- First coefficient of the `n`-th power of a power series. -/
lemma coeff_one_pow (n : ℕ) (φ : R⟦X⟧) :
    coeff R 1 (φ ^ n) = n * coeff R 1 φ * (constantCoeff R φ) ^ (n - 1) := by
  rcases Nat.eq_zero_or_pos n with (rfl | hn)
  · simp
```

**Comment by riccardobrasca** [comment]
You also have a conflict... otherwise LGTM, thanks!

bors d+

**Comment by rwst** [comment]
That conflict is not really there. I merged master without change, and CI builds Mathlib without problems.

**Comment by riccardobrasca** [comment]
> That conflict is not really there. I merged master without change, and CI builds Mathlib without problems.

Did you pull before merging? It says it is in ` Mathlib/RingTheory/PowerSeries/Basic.lean `.

**Comment by rwst** [comment]
> > That conflict is not really there. I merged master without change, and CI builds Mathlib without problems.
> 
> Did you pull before merging? It says it is in `Mathlib/RingTheory/PowerSeries/Basic.lean`.

I have edited it in browser.

**Pull request closed without merging**